code/game/world.dm

```diff
@@ -139,6 +139,216 @@ GLOBAL_VAR(tracy_log)
+			return TRUE
+
+/// Inserts our current world.cpu value into our rolling lists
+/// Its job is to pull the actual usage last tick instead of the moving average
+/world/proc/unroll_cpu_value()
+	if(GLOB.last_cpu_update == world.time)
+		return
+	GLOB.last_cpu_update = world.time
+	// cache for sonic speed
+	var/list/cpu_values = GLOB.cpu_values
+	var/list/avg_cpu_values = GLOB.avg_cpu_values
+	var/cpu_index = GLOB.cpu_index
+	var/avg_cpu = world.cpu
+	// We need to hook into the INSTANT we start our moving average so we can reconstruct gained/lost cpu values
+	// Defaults to null or 0 so the wrap here is safe for the first 16 entries
+	var/lost_value = cpu_values[WRAP(cpu_index - WINDOW_SIZE, 1, CPU_SIZE + 1)]
+
+	// ok so world.cpu is a 16 entry wide moving average of the actual cpu value
+	// because fuck you
+	// I want the ACTUAL unrolle value, so I need to deaverage it. this is possible because we have access to ALL values and also math
+	// yes byond does average against a constant window size, it doesn't account for a lack of values initially it just sorta assumes they exist.
+	// ♪ it ain't me, it ain't me ♪
+
+	// Second tick example
+	// avg = (A + B) / 4
+	// old_avg = (A) / 4
+	// (avg * 4 - old_avg * 4) roughly sans floating point BS = B
+	// Fifth tick example
+	// avg = (B + C + D + E) / 4
+	// old_avg = (A + B + C + D) / 4
+	// (avg * 4 - old_avg * 4) roughly = E - A
+	// so after we start losing numbers we need to add the one we're losing
+	// We're trying to do this with as few ops as possible to avoid noise
+	// soooo
+	// E = (avg * 4 - old_avg * 4) + A
+
+	var/last_avg_cpu = avg_cpu_values[WRAP(cpu_index - 1, 1, CPU_SIZE + 1)]
+	var/real_cpu = avg_cpu * WINDOW_SIZE - last_avg_cpu * WINDOW_SIZE + lost_value
+
+	// due to I think? compounded floating point error either on our side or internal to byond we somtimes get way too large/small cpu values
+	// I can't correct in place because I need the full history of averages to add back lost values
+	// our cpu value for last tick cannot be lower then the cost of sleeping procs + map cpu, so we'll clamp to that
+	// my hope is this will keep error within a reasonable bound as storing a lower then expected number would cause a higher then expected number as a side effect
+	var/lower_bound = GLOB.tick_cpu_usage[cpu_index] + world.map_cpu
+
+	cpu_values[cpu_index] = max(real_cpu, lower_bound)
+	avg_cpu_values[cpu_index] = avg_cpu
+	GLOB.map_cpu_usage[cpu_index] = world.map_cpu
+	GLOB.verb_cost[cpu_index] = max(real_cpu - lower_bound, 0)
+	GLOB.cpu_index = WRAP(cpu_index + 1, 1, CPU_SIZE + 1)
+	GLOB.cpu_tracker.update_display()
+	// make an animated display of cpu usage to get a better idea of how much we leave on the table
+
+/proc/update_glide_size()
+	world.unroll_cpu_value()
+	var/list/cpu_values = GLOB.cpu_values
+	var/sum = 0
+	var/non_zero = 0
+	#warn there's gotta be a better way then this, or at least a less shitty way
+	for(var/value in cpu_values)
+		sum += max(value, 100)
```

Comment:
why clip it?

Reply:
to not drag the average down, since what we actually want to average here is the LAG, rather then the %